__tests__/app.test.js

```diff
@@ -168,6 +169,50 @@ describe('/api/articles', () =>
                         });
                 });
             });
+
+            describe('POST', () =>
+            {
+                test("STATUS 201 - Responds with the posted comment object.", () =>
+                {
+                    return request(app)
+                        .post('/api/articles/1/comments')
+                        .send({ username: 'butter_bridge', body: 'This is a test comment.' })
+                        .expect(201)
+                        .then(({ body: { comment } }) =>
+                        {
+                            expect(comment).toMatchObject(
+                                {
+                                    comment_id: 19,
+                                    votes: 0,
+                                    author: 'butter_bridge',
+                                    body: 'This is a test comment.',
+                                    article_id: 1
+                                }
+                            );
+                            expect(comment.created_at).toBeString(); // Dynamic test - separated due to 'created_at' being the current time of posting
+                        });
+                });
+                test("STATUS 400 - Responds with 'Bad Request' when posted object is missing required properties.", () => {
+                    return request(app)
+                        .post('/api/articles/1/comments')
+                        .send({ body: 'This is a test comment.' })
+                        .expect(400)
+                        .then(({ body: { msg } }) =>
+                        {
+                            expect(msg).toBe('Bad Request');
+                        });
+                });
+                test("STATUS 400 - Responds with 'Bad Request' when posted object has invalid property values.", () => {
+                    return request(app)
+                        .post('/api/articles/1/comments')
+                        .send({ username: 'not_an_existing_user', body: 'This is a test comment.' })
```

Comment:
non-existing username would give 404 - username does not exist instead of 400 bad request